ReviewChecklist.md

```diff
@@ -0,0 +1,167 @@
+
+### Common Pitfalls
+The following points are all covered by the coding style but come up so often that it is worth singling them out here:
+- [ ] **Always initialize value types in the definition,** even if you are sure you will assign them later.
+- [ ] **Use "east const" style.** I.e. `T const*`, not `const T *`.
+- [ ] **Keep indentation consistent.** See our [`.editorconfig`](.editorconfig).
+    - [ ] Tabs for C++. But use them for indentation only. Any whitespace later on the line must consist of spaces.
+    - [ ] 4 spaces for most other file types.
+- [ ] **Use `auto` sparingly.** Only use it when the actual type is very long and complicated or when it is
+    already used elsewhere in the same expression.
+- [ ] **Indent braces and parentheses in a way that makes nesting clear.**
+- [ ] **Use `using namespace` only in `.cpp` files.** Use it for `std` and our own modules.
+    Avoid unnecessary `std::` prefix in `.cpp` files (except for `std::move`).
+- [ ] **Use range-based loops and destructuring.**
+- [ ] **Include any headers you use directly,** even if they are implicitly included through other headers.
+
+## Documentation
+- [ ] **Does the PR update relevant documentation?**
+
+### Documentation Style and Good Practices
+- [ ] **Use double backticks in RST (``` ``x`` ```). Prefer single backticks in Markdown (`` `x` ``),**
+    but note that double backticks are valid too and we use them in some cases for legacy reasons.
+- [ ] **Always start a new sentence on a new line.**
+    This way you do not have to rewrap the surrounding text when you rewrite the sentence.
+    This also makes changes actually easier to spot in the diff.
+
+## Testing
+
+### What to Test
+- [ ] **Is newly added code adequately covered by tests?** Have you considered all the important corner cases?
+- If it is a bugfix:
+    - [ ] **Is the (cleaned up) repro from the issue included as a test case?**
+    - [ ] **Do tests cover not just the bug repro but also related cases that might be affected by a similar bug?**
+- [ ] **Positive cases (code that compiles) should have a semantic test.**
+- [ ] **Negative cases (code with compilation errors) should have a syntax test.**
+- [ ] **Avoid mixing positive and negative cases in the same syntax test.**
+    If the test produces an error, we stop at the analysis stage and we will not detect
+    problems that only occur in code generation, optimizer or assembler.
+    - [ ] If you have to do it, at least mark positive cases inside the file with a short comment.
+        - This way, when the test is updated, it is easier to verify that these cases still do not trigger an error.
+- [ ] New syntax: **does it have an [`ASTJSON`](test/libsolidity/ASTJSON/) test?**
+- [ ] New CLI or StandardJSON option:
+    - [ ] **Does it have a [command-line test](test/cmdlineTests/)?**
+    - [ ] **Is the option listed for every input mode in [`CommandLineParser` tests](test/solc/CommandLineParser.cpp)?**
```

Comment:
`Add option to every input mode in CommandLineParser..`

Reply:
Same as https://github.com/ethereum/solidity/pull/13553#discussion_r982889983.